### PR TITLE
fix: search component improvement

### PR DESCRIPTION
**Pull request by blakmetall**
* fix: 'recentItem' code moved to a component
* fix: eslint update -- return default forbid elements and instead checkContextTypes was set to true
* global styled added for shared styles
* eslint update -- import/prefer-default-export -- updated to remove lint error.
styled component with single export was throwing a lint error

@nexxtway/rainbow-algolia-search

### CODE REVIEW

**Comment by blakmetall** [comment]
I moved the components for the SerachInput and RecentItem to components folder.

This might be the last push to this PR

File: src/components/recent-item/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { RecentItemLink } from './styled';
+
+const recentItem = 'You Don’t Know JS';
+
+const RecentItem = () => {
+    if (recentItem) {
+        return (
+            <InputCallToAction>
+                Recent Book:
+                <RecentItemLink href="#">{recentItem}</RecentItemLink>
```
Comment (LeandroTorresSicilia) [inline src/components/recent-item/index.js]: make recentItem => text a prop

File: src/components/search-input/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { InputWrapper } from './styled';
+
+const SearchInput = () => {
+    return <InputWrapper isCentered placeholder="Find a Book" />;
```
Comment (LeandroTorresSicilia) [inline src/components/search-input/index.js]: why not use InputWrapper direclty

File: src/components/search-input/styled.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import styled from 'styled-components';
+import { Input } from 'react-rainbow-components';
+
+export const InputWrapper = styled(Input)`
```
Comment (LeandroTorresSicilia) [inline src/components/search-input/styled.js]: move this styled to the styled of home page

File: src/styled.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import styled from 'styled-components';
+
+export const InputCallToAction = styled.div`
```
Comment (LeandroTorresSicilia) [inline src/styled.js]: change this name

**Pull request merged**